packages/flutter_tools/lib/src/version.dart

```diff
@@ -111,28 +112,29 @@ class FlutterVersion {
           globals.processUtils,
           _workingDirectory,
         );
-        _channel = channel.substring(slash + 1);
+        channel = channel.substring(slash + 1);
       } else if (channel.isEmpty) {
-        _channel = 'unknown';
+        channel = 'unknown';
       } else {
-        _channel = channel;
+        channel = channel;
       }
+      _channel = channel;
     }
-    return _channel;
+    return channel;
```

Comment:
does this not need a cast from `String?` to `String`?

Reply:
> does this not need a cast from `String?` to `String`?

No, there's no logical branch that sets it to a nullable value.  See more explanation in https://github.com/flutter/flutter/pull/79911#discussion_r609054707